x/rollapp/types/chain_id.go

```diff
@@ -43,20 +46,28 @@ func NewChainID(id string) (ChainID, error) {
 	if err != nil {
 		return ChainID{}, err
 	}
+	revision, err := getRevisionNumber(chainID)
+	if err != nil {
+		return ChainID{}, err
+	}
+	matches := strings.Split(chainID, "-")
```

Comment:
if we add this test case:

```
{
	name:      "starts with dash",
	rollappId: "-1234",
	eip:       false,
	valid:     false,
}
```

to the `TestCreateRollappId` test, the resulting `ChainID` will be:

```
ChainID {
	chainID: "-1234",
	eip155ID: nil,
	revision: 1234,
	name:     "",
}
```

Reply:
And yeah, the test will fail, an error will be expected, but there won't be any

Reply:
empty name fix added